### PR TITLE
Slider: convert tests to use testing-library

### PR DESCRIPTION
**Pull request by TristanWatanabe**
## Current Behavior

- v8 `Slider` tests don't use testing-library

## New Behavior

- Convert Slider to fully use testing-library
## Related Issue(s)

part of #21663

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/d8ab052dcb2a452d879f9a01ae8abd30fd44f57b'>d8ab052dcb2a452d879f9a01ae8abd30fd44f57b</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Avatar-mount_base.html">886</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Avatar-mount.html">929</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>BaseButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/BaseButton-mount_base.html">967</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/BaseButton-mount.html">941</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Breadcrumb</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Breadcrumb-mount_base.html">2740</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Breadcrumb-mount.html">2740</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ButtonNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ButtonNext-mount_base.html">505</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ButtonNext-mount.html">508</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Checkbox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Checkbox-mount_base.html">1596</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Checkbox-mount.html">1587</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>CheckboxBase</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/CheckboxBase-mount_base.html">1353</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/CheckboxBase-mount.html">1371</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ChoiceGroup</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ChoiceGroup-mount_base.html">4895</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ChoiceGroup-mount.html">4969</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ComboBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ComboBox-mount_base.html">1026</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ComboBox-mount.html">997</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>CommandBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/CommandBar-mount_base.html">10353</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/CommandBar-mount.html">10489</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>ContextualMenu</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ContextualMenu-mount_base.html">8688</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ContextualMenu-mount.html">8742</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DefaultButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DefaultButton-mount_base.html">1159</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DefaultButton-mount.html">1158</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRow</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DetailsRow-mount_base.html">3927</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DetailsRow-mount.html">3845</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowFast</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DetailsRowFast-mount_base.html">3796</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DetailsRowFast-mount.html">3858</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>DetailsRowNoStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DetailsRowNoStyles-mount_base.html">3575</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DetailsRowNoStyles-mount.html">3651</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Dialog</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Dialog-mount_base.html">2301</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Dialog-mount.html">2293</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>DocumentCardTitle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DocumentCardTitle-mount_base.html">210</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/DocumentCardTitle-mount.html">175</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Dropdown</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Dropdown-mount_base.html">3324</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Dropdown-mount.html">3344</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderNext</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderNext-mount_base.html">1912</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderNext-mount.html">1968</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderWithTheme-mount_base.html">170</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderWithTheme-mount.html">166</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderWithTheme-virtual-rerender_base.html">111</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderWithTheme-virtual-rerender.html">104</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">205</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FluentProviderWithTheme-virtual-rerender-with-unmount.html">207</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FocusTrapZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FocusTrapZone-mount_base.html">1916</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FocusTrapZone-mount.html">1908</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FocusZone</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FocusZone-mount_base.html">1913</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/FocusZone-mount.html">1873</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>IconButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/IconButton-mount_base.html">1788</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/IconButton-mount.html">1774</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Label</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Label-mount_base.html">396</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Label-mount.html">378</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Layer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Layer-mount_base.html">3094</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Layer-mount.html">3100</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Link</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Link-mount_base.html">517</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Link-mount.html">508</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/MakeStyles-mount_base.html">1771</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/MakeStyles-mount.html">1735</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>MenuButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/MenuButton-mount_base.html">1555</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/MenuButton-mount.html">1516</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MessageBar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/MessageBar-mount_base.html">2064</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/MessageBar-mount.html">2118</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Nav</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Nav-mount_base.html">3394</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Nav-mount.html">3389</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>OverflowSet</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/OverflowSet-mount_base.html">1167</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/OverflowSet-mount.html">1166</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Panel</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Panel-mount_base.html">2249</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Panel-mount.html">2217</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Persona-mount_base.html">900</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Persona-mount.html">904</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>Pivot</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Pivot-mount_base.html">1464</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Pivot-mount.html">1502</a></td>
            <td>1000</td>
            <td></td>
           </tr>
<tr>
            <td>PrimaryButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/PrimaryButton-mount_base.html">1344</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/PrimaryButton-mount.html">1354</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Rating</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Rating-mount_base.html">7901</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Rating-mount.html">7926</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SearchBox</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SearchBox-mount_base.html">1408</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SearchBox-mount.html">1420</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Shimmer</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Shimmer-mount_base.html">2635</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Shimmer-mount.html">2648</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Slider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Slider-mount_base.html">2058</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Slider-mount.html">2073</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SpinButton-mount_base.html">5212</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SpinButton-mount.html">5170</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Spinner</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Spinner-mount_base.html">454</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Spinner-mount.html">474</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SplitButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SplitButton-mount_base.html">3262</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SplitButton-mount.html">3253</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Stack</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Stack-mount_base.html">576</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Stack-mount.html">569</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithIntrinsicChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/StackWithIntrinsicChildren-mount_base.html">2330</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/StackWithIntrinsicChildren-mount.html">2358</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>StackWithTextChildren</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/StackWithTextChildren-mount_base.html">5355</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/StackWithTextChildren-mount.html">5444</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SwatchColorPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SwatchColorPicker-mount_base.html">11730</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/SwatchColorPicker-mount.html">11758</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TagPicker</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/TagPicker-mount_base.html">2694</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/TagPicker-mount.html">2704</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TeachingBubble</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/TeachingBubble-mount_base.html">13497</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/TeachingBubble-mount.html">13417</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Text</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Text-mount_base.html">474</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Text-mount.html">454</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>TextField</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/TextField-mount_base.html">1447</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/TextField-mount.html">1452</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ThemeProvider-mount_base.html">1248</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ThemeProvider-mount.html">1231</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ThemeProvider-virtual-rerender_base.html">675</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ThemeProvider-virtual-rerender.html">648</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>ThemeProvider</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ThemeProvider-virtual-rerender-with-unmount_base.html">1964</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/ThemeProvider-virtual-rerender-with-unmount.html">1912</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Toggle</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Toggle-mount_base.html">827</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/Toggle-mount.html">861</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>buttonNative</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/buttonNative-mount_base.html">170</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/21919/perf-test/buttonNative-mount.html">173</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9423"/>

File: packages/react/src/components/Slider/Slider.test.tsx [hunk 1]
```diff
@@ -1,9 +1,6 @@
 import * as React from 'react';
-import * as ReactDOM from 'react-dom';
-import * as ReactTestUtils from 'react-dom/test-utils';
-import { mount, ReactWrapper } from 'enzyme';
-import { resetIds, KeyCodes } from '@fluentui/utilities';
-import { create } from '@fluentui/utilities/lib/test';
+import { resetIds, KeyCodes } from '../../Utilities';
+import { fireEvent, render } from '@testing-library/react';
```
Comment (ecraig12345) [inline packages/react/src/components/Slider/Slider.test.tsx:3]: This is going to be in the testing guidance, but it's generally recommended to use userEvent rather than fireEvent, since it more realistically simulates the full sequence of events triggered by user interaction. https://www.npmjs.com/package/@testing-library/user-event

File: packages/react/src/components/Slider/Slider.test.tsx [hunk 2]
```diff
@@ -34,95 +25,94 @@ describe('Slider', () => {
...
+    const { container, getByRole } = render(<Slider id="test_id" styles={{ titleLabel: 'test_label' }} />);
 
-    const sliderSlideBox = wrapper.find('.ms-Slider-slideBox');
-    expect(sliderSlideBox.getDOMNode().id).toEqual('test_id');
+    expect(getByRole('slider').id).toEqual('test_id');
 
     // properly associates label with custom id
-    const label = wrapper.find('label.test_label');
-    expect(label.prop('htmlFor')).toBe('test_id');
+    const label = container.getElementsByClassName('test_label')[0];
```
Comment (ecraig12345) [inline packages/react/src/components/Slider/Slider.test.tsx:62]: I think it might be slightly better/more idiomatic to get the label by role, then test its class and for. Something like this (haven't run this in TS and might be remembering property names incorrectly).
```ts
const label = getByRole('label');
expect(label.className).toMatch(/test_label/);
expect(label.getAttribute('for')).toBe('test_id');
```

And similar in the next test.

File: packages/react/src/components/Slider/Slider.test.tsx [hunk 3]
```diff
@@ -34,95 +25,94 @@ describe('Slider', () => {
...
-    wrapper = mount(<Slider defaultValue={0} min={-100} max={100} componentRef={slider} />);
+    render(<Slider defaultValue={0} min={-100} max={100} componentRef={slider} />);
     expect(slider.current!.value).toEqual(0);
   });
 
   it('handles zero value', () => {
     const slider = React.createRef<ISlider>();
 
-    wrapper = mount(<Slider value={0} min={-100} max={100} componentRef={slider} />);
+    render(<Slider value={0} min={-100} max={100} componentRef={slider} />);
```
Comment (ecraig12345) [inline packages/react/src/components/Slider/Slider.test.tsx:89]: Not a new issue but it would be nice if we could test something in the DOM to make sure the correct value is rendered as well. Since this implementation doesn't use a native input, you'd have to look if there are any aria attributes or something that you could check.

File: packages/react/src/components/Slider/Slider.test.tsx [hunk 4]
```diff
@@ -34,95 +25,94 @@ describe('Slider', () => {
...
     expect(slider.current!.value).toEqual(0);
   });
 
   it('calls onChange and onChanged when slider value changes with mouse', () => {
     const onChange = jest.fn();
     const onChanged = jest.fn();
     const slider = React.createRef<ISlider>();
-    wrapper = mount(<Slider onChange={onChange} defaultValue={5} onChanged={onChanged} componentRef={slider} />);
+    const { container, getByRole } = render(
```
Comment (ecraig12345) [inline packages/react/src/components/Slider/Slider.test.tsx:97]: I wonder if this would be a good candidate for doing an additional test in cypress to ensure it works in real scenarios as well. But that's probably lower priority if the jest test is reliable.

File: packages/react/src/components/Slider/Slider.test.tsx [hunk 5]
```diff
@@ -34,95 +25,94 @@ describe('Slider', () => {
...
+      <Slider onChange={onChange} defaultValue={5} onChanged={onChanged} componentRef={slider} />,
+    );
 
-    const sliderLine = wrapper.find('.ms-Slider-line');
-    const sliderThumb = wrapper.find('.ms-Slider-slideBox');
+    const sliderLine = container.getElementsByClassName('ms-Slider-line')[0];
```
Comment (ecraig12345) [inline packages/react/src/components/Slider/Slider.test.tsx:101]: Generally getting things by class isn't recommended in testing-library (because it's not how the user would interact with the control) but since this one is weird and doesn't use a regular input, there might not be a way around it.

File: packages/react/src/components/Slider/Slider.test.tsx [hunk 6]
```diff
@@ -259,84 +250,74 @@ describe('Slider', () => {
...
-    sliderThumb.simulate('mousedown', { type: 'mousedown', clientX: 10, clientY: 0 });
-
-    sliderThumb.simulate('mousedown', { type: 'mousedown', clientX: 20, clientY: 0 });
+    fireEvent.mouseDown(sliderThumb, { clientX: 10, clientY: 0 });
+    fireEvent.mouseDown(sliderThumb, { clientX: 20, clientY: 0 });
 
     expect(range).toEqual([2, 5]);
   });
 
-  it('has type=button on all buttons', () => {
```
Comment (ecraig12345) [inline packages/react/src/components/Slider/Slider.test.tsx:277]: Why was this removed?

**Review by ecraig12345 — COMMENTED** [review]
Looking good--thanks for doing this! A few suggestions.

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 1 above)
Comment (TristanWatanabe) [reply packages/react/src/components/Slider/Slider.test.tsx:3]: I can port some of the mouseclick behaviors to use `userEvent` but there are some tests (such as checking if `onChange` triggers on a `mousedown` click while `onChanged` does not) where using `fireEvent.mousedown` was needed. Also, for the keyboard tests, using `userEvent.type(element, '{arrowdown}')` syntax to simulate a keydown was throwing the error below which is why I settled on using `fireEvent.keydown(element, {keyCode})`:

image

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 2 above)
Comment (TristanWatanabe) [reply packages/react/src/components/Slider/Slider.test.tsx:62]: Yup that's what I initially had in mind but apparently the label element doesn't have a corresponding aria role
image

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 6 above)
Comment (TristanWatanabe) [reply packages/react/src/components/Slider/Slider.test.tsx:277]: This test didn't do anything since the Slider component doesn't have any buttons

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 5 above)
Comment (TristanWatanabe) [reply packages/react/src/components/Slider/Slider.test.tsx:101]: Yup this one was tricky but that's good to know about not getting things by class!

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 3 above)
Comment (TristanWatanabe) [reply packages/react/src/components/Slider/Slider.test.tsx:89]: If I'm understanding this correctly, I think this should be covered by the `renders correct aria-valuetext` test below

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 6 above)
Comment (ecraig12345) [reply packages/react/src/components/Slider/Slider.test.tsx:277]: Ha, wow... 😆

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 3 above)
Comment (ecraig12345) [reply packages/react/src/components/Slider/Slider.test.tsx:89]: Right, I guess it just concerns me somewhat that we're checking the internal component state without checking what it's actually rendering (in case there's a mismatch)

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 1 above)
Comment (ecraig12345) [reply packages/react/src/components/Slider/Slider.test.tsx:3]: Oh interesting, I bet it's because it tries to do a click first. The docs mention there's a `skipClick` option--what happens if you use that?
> type will click the element before typing. To disable this, set the skipClick option to true.

Comment (TristanWatanabe) [reply packages/react/src/components/Slider/Slider.test.tsx:3]: Nevermind on this error, just needed to change my approach on testing it

File: packages/react/src/components/Slider/Slider.test.tsx (hunk 2 above)
Comment (ecraig12345) [reply packages/react/src/components/Slider/Slider.test.tsx:65]: Oh interesting...I didn't see any other suggestions from a brief online search, so I guess we can stick with this.

**Review by ecraig12345 — APPROVED** [review]
